DEVINFO.md

```diff
@@ -0,0 +1,174 @@
+# How to Create and Review PRs
+
+## Overview
+
+Blockchain software development requires a much higher degree of rigor than most other kinds of software.  This is because with blockchains, **there is no roll-back** from a bad deployment.  There is essentially zero room for consensus bugs.  If you ship a consensus bug, that bug could not only have catastrophic consequences for users (i.e. they lose all their money), but also be intractable to fix, mitigate, or remove.  This is because unlike nearly every other kind of networked software, **the state of the blockchain is what the users' computers say it is.**  If you want to make changes, you _must_ get _user_ buy-in, and this is necessarily time-consuming and not at all guaranteed to succeed.
```

Comment:
I'd slim this down a lot:

```suggestion
Blockchain software development requires a much higher degree of rigor than most other kinds of software.  This is because with blockchains, **there is no roll-back** from a bad deployment.
```

I think the PR reviewing guide's job is to tell people how to review in this repo, not really educate on blockchain development consequences and I think that the shorter these kinds of docs are, the more likely they are to be read and understood.

Reply:
I still think it's important to keep this content somewhere.  By far the biggest intellectual stumbling block I've seen from new contributors is that **they do not appreciate the gravity of the situation** they find themselves in.  We can't afford for them to make mistakes, and the best way to address this is to be up-front about the extremely high cost of failure.  My hope is that developing this expectation early on will motivate them to stick it out through what will likely be several rounds of review on their first PRs.

Reply:
I'm going to merge and incorporate this file into `/docs/CONTRIBUTING.md` (and remove our `/CONTRIBUTORS.md` file), I'll add the content to the top of that file.